### ISSUE TITLE
Create custom EventHandler for UserControl in WinUI3

### ISSUE THREAD

**Issue by AtosNicoS** [microsoft/microsoft-ui-xaml#7660]
Hi!
I simply want to add an EventHandler to my UserControl that I can then reference in the xaml, whenever I use the usercontrol. Cimilar to the Button `Click` Event. However that does not work.

The only [stackoverflow](https://stackoverflow.com/questions/24642334/exposing-event-in-custom-usercontrol-window-store-app) post out there also did not help, I tried all kind of combinations. None worked. The event is not recognized in xaml.

How do I do this? is it a bug?

```c#
public EventHandler<IList<Guid>>? MyCustomEvent;
public EventHandler? MyCustomEvent2;
public TypedEventHandler<MapPanel, IList<Guid>>? MyCustomEvent3;
public RoutedEventHandler? MyCustomEvent4;

public EventHandler<IList<Guid>>? MyCustomEvent5
{
	get => (EventHandler<IList<Guid>>?)GetValue(MyCustomEvent5Property);
	set => SetValue(MyCustomEvent5Property, value);
}
public static readonly DependencyProperty MyCustomEvent5Property =
	DependencyProperty.Register(nameof(MyCustomEvent5), typeof(EventHandler<IList<Guid>>),
		typeof(MapPanel), new PropertyMetadata(null));
```

**Comment by DarranRowe** [comment]
The Xaml compiler reads metadata so it should be able to just hook it up without a dependency property.
When I author a user control in a C++ WinUI application, all I have is:
```
[default_interface]
runtimeclass SillyControl : Microsoft.UI.Xaml.Controls.UserControl
{
	SillyControl();

	event Windows.Foundation.EventHandler<Object> ControlLoaded;
}
```
for the component definition and the Xaml compiler picks it up and uses it without issue.
This would suggest that the issue is that the Xaml compiler isn't seeing what you are doing as events. But C# does have the event keyword, so have you tried using this? As an example:
`
public event EventHandler<Object> MyCustomEvent;
`

**Comment by castorix** [comment]
You can see in the MediaEngine sample, in [CImageButton.xaml.cs](https://github.com/castorix/WinUI3_MediaEngine/blob/master/CImageButton.xaml.cs),
I added an event to a UserControl, named _CImageButtonClicked_,  and it appears in XAML : 

![CImageButtonClicked](https://user-images.githubusercontent.com/22345506/187378479-28a6bde2-53c0-4ca7-b7dc-e55dc4aeaecc.jpg)

**Comment by AtosNicoS** [comment]
Hey guys, I found my issue finally.

Thats is what I tried:
```c#
public EventHandler<IList<Guid>>? MyCustomEvent;
public EventHandler? MyCustomEvent2;
```

And this is what works:
```c#
public event EventHandler<IList<Guid>>? MyCustomEvent;
public event EventHandler? MyCustomEvent2;
```

I  forgot to add the `event` in front of the `EventHandler`. Not that the questionsmark `?` are used, because I've [enabled the nullable feature](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types) in this winui3 project.
